### PR TITLE
Templates

### PR DESCRIPTION
**Pull request by hzoo**

### CODE REVIEW

**Comment by hzoo** [comment]
How can we check the `TemplateElement` `tail` property? Have to look to the next (non-children) token and see if it's ``` ` ```?

**Comment by hzoo** [comment]
Never mind doing `quasis[quasis.length - 1].tail = true;` works for now.

File: src/elementTree.js [hunk 1]
```diff
@@ -66,9 +66,9 @@ const visitorKeys = {
     /*IT*/ Super: [],
     /*IT*/ SwitchStatement: ['discriminant', 'cases'],
     /*IT*/ SwitchCase: ['test', 'consequent'],
-    /*--*/ TaggedTemplateExpression: ['tag', 'quasi'],
-    /*--*/ TemplateElement: [],
-    /*--*/ TemplateLiteral: ['quasis', 'expressions'],
+    /*IT*/ TaggedTemplateExpression: ['tag', 'quasi'],
+    /*IT*/ TemplateElement: ['tail', 'value'],
```
Comment (hzoo) [inline src/elementTree.js]: added these props

**Comment by hzoo** [comment]
Trying to do a nested template

Kind of confused as to why the nodes aren't changing to be like a BinaryExpression

for `parseAndGetExpression('`b${`${1+1}c`}3`');`
```
[ 'Punctuator: `',
  'TemplateElement',
  'Punctuator: ${',
  'Punctuator: `',
  'Punctuator: template',
  'Punctuator: ${',
  'Numeric',
  'Punctuator: +',
  'Numeric',
  'Punctuator: }',
  'Punctuator: c',
  'Punctuator: `',
  'Punctuator: }',
  'TemplateElement',
  'Punctuator: `' ]
    1) should accept nested templates


  0 passing (6ms)
  1 failing

  1) TemplateLiteral should accept nested templates:
     Error: Expected token type "Punctuator" but "Numeric" found
```

File: src/elementTree.js (hunk 1 above)
Comment (mdevils) [inline src/elementTree.js]: These are not needed (we only store props which contain AST nodes here).

**Pull request merged**